task/src/main/java/com/flipkart/flux/impl/task/AkkaTask.java

```diff
@@ -131,10 +132,30 @@ public void onReceive(Object message) throws Exception {
                                 new FluxError.ExecutionContextMeta(taskAndEvent.getStateMachineId(), taskAndEvent.getTaskId(),
                                         taskAndEvent.getRetryCount(), taskAndEvent.getCurrentRetryCount()));
                     } else {
-                        // mark the task outcome as execution failure
-                        fluxRuntimeConnector.updateExecutionStatus(
-                                new ExecutionUpdateData(taskAndEvent.getStateMachineId(), taskAndEvent.getTaskId(), Status.errored,
-                                        taskAndEvent.getRetryCount(), taskAndEvent.getCurrentRetryCount(), hre.getMessage()));
+                        //check if the exception hierarchy has FluxRetriableException, if yes trigger retry
+                        boolean isFluxRetriableException = false;
+                        Throwable cause = hre.getCause();
+                        while(cause != null) {
+                            if(cause.getClass().getName().equals(FluxRetriableException.class.getName())) {
```

Comment:
It will not work, if the task throws an exception of type that extends FluxRetriableException.